src/main/java/ch/njol/skript/lang/SkriptParser.java

```diff
@@ -225,7 +220,23 @@ public boolean hasTag(String tag) {
 								}
 							}
 							T element = info.getElementClass().newInstance();
-							if (element.init(parseResult.exprs, patternIndex, getParser().getHasDelayBefore(), parseResult)) {
+
+							if (element instanceof EventRestrictedSyntax eventRestrictedSyntax) {
+								Class<? extends Event>[] supportedEvents = eventRestrictedSyntax.supportedEvents();
+								if (!getParser().isCurrentEvent(supportedEvents)) {
+									StringJoiner joiner = new StringJoiner(", ", "the ", "");
```

Comment:
Does this properly handle multiple supported events? specifically, using or/and:
``x can only be used in the a, b, or c events``